### PR TITLE
meeting-briefing: Agentic approach with research tools

### PR DESCRIPTION
**Pull request by elie222**
# User description
Meeting Briefing: Refactor to agentic approach with research tools

Refactored the meeting briefing system from a pipeline approach to an agentic one, where the AI writes the full briefing with context-aware tool access.

- Converted `aiGenerateMeetingBriefing` to use `generateText` with `researchGuest` and `finalizeBriefing` tools.
- Simplified `gatherContextForEvent` to remove parallel research calls.
- Added comprehensive AI and unit tests in `ai-meeting-briefing.test.ts`.
- Improved prompt coherence by giving the AI full context of all guests simultaneously.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for AI meeting briefing flows covering formatting, per-guest bullets, past meetings, email history, edge cases, and end-to-end scenarios.

* **New Features**
  * Meeting briefs now deliver structured per-guest bullet summaries.
  * Email briefings include an AI-generated accuracy disclaimer for names.

* **Refactor**
  * Brief generation moved to a tool-driven agent flow and simplified context gathering (removed prior per-guest AI research).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Refactors the meeting briefing system to an agentic, tool-driven approach, converting <code>aiGenerateMeetingBriefing</code> to utilize <code>generateText</code> with <code>researchGuest</code> and <code>finalizeBriefing</code> tools. Simplifies context gathering by removing parallel research calls and improves prompt coherence, alongside adding comprehensive AI and unit tests.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1162?tool=ast&topic=Agentic+Briefing+Workflow>Agentic Briefing Workflow</a>
        </td><td>Refactors the meeting briefing generation to an agentic, tool-driven workflow, converting <code>aiGenerateMeetingBriefing</code> to use <code>generateText</code> with <code>researchGuest</code> and <code>finalizeBriefing</code> tools, and updating the <code>AGENTIC_SYSTEM_PROMPT</code> for conditional tool usage and improved coherence.<details><summary>Modified files (1)</summary><ul><li>apps/web/utils/ai/meeting-briefs/generate-briefing.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>adjust-how-we-do-resea...</td><td>January 01, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1162?tool=ast&topic=AI+Briefing+Tests>AI Briefing Tests</a>
        </td><td>Adds comprehensive AI and unit tests for the <code>aiGenerateMeetingBriefing</code> function, covering various scenarios including single guests with no prior context and ensuring proper mock clearing before each test.<details><summary>Modified files (1)</summary><ul><li>apps/web/__tests__/ai-meeting-briefing.test.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>fixes</td><td>January 01, 2026</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1162?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/__tests__/ai-meeting-briefing.test.ts [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+  test("includes no_prior_context tag for guests without history", () => {
+    const data = getMeetingBriefingData({
+      externalGuests: [{ email: "new@contact.com", name: "New Contact" }],
+      emailThreads: [],
+      pastMeetings: [],
+    });
+    const prompt = buildPrompt(data, null);
+
+    expect(prompt).toContain("<no_prior_context>");
+    expect(prompt).toContain("new contact");
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/ai-meeting-briefing.test.ts]: Fixed the test assertion to correctly verify the guest's name 'New Contact'.

File: apps/web/__tests__/ai-meeting-briefing.test.ts [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+  console.log(`Meeting: ${meetingTitle}`);
+  console.log("\nGuests:");
+  for (const guest of result.guests) {
+    console.log(`\n  ${guest.name} (${guest.email})`);
+    for (const bullet of guest.bullets) {
+      console.log(`    - ${bullet}`);
+    }
+  }
+  console.log(`${"=".repeat(80)}\n`);
+}
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/ai-meeting-briefing.test.ts:496]: Replaced console.log with console.debug in prettyPrintBriefing as per LLM testing guidelines.

File: apps/web/__tests__/ai-meeting-briefing.test.ts [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+      // Add minimal email context so test doesn't rely solely on research API
+      const mockMessage = getMockMessage({
+        from: "new.person@example.com",
+        to: "user@test.com",
+        subject: "Looking forward to our coffee chat",
+        textPlain:
+          "Hi! Excited to meet tomorrow. I work as a product manager at TechCo.",
+      });
+
+      const data = getMeetingBriefingData({
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/ai-meeting-briefing.test.ts:209]: Good catch! Added `beforeEach(() => vi.clearAllMocks())` to the AI test suite to prevent mock state leakage between tests.

File: apps/web/utils/ai/meeting-briefs/generate-briefing.ts [hunk 4]
```diff
@@ -22,55 +38,307 @@ const briefingSchema = z.object({
...
+  // Build tools based on what's configured
+  const tools = buildSearchTools({ emailAccount, logger });
+
+  // If no search tools are available, skip the agentic approach
+  if (Object.keys(tools).length === 0) {
+    logger.info(
+      "No search tools configured, skipping agentic briefing generation",
+    );
+    return generateFallbackBriefing(briefingData.externalGuests);
+  }
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/ai/meeting-briefs/generate-briefing.ts:95]: Great catch! Fixed this regression - now when no search tools are configured, the LLM still runs with the existing email/meeting context and just the `finalizeBriefing` tool. The agent can summarize what we already know about the guest rather than returning a generic fallback. Updated the system prompt to clarify that search tools are optional.

File: apps/web/utils/ai/meeting-briefs/generate-briefing.ts [hunk 5]
```diff
@@ -103,8 +384,12 @@ ${event.description ? `Description: ${event.description}` : ""}
 <guest_context>
 ${guestContexts.map((guest) => formatGuestContext(guest)).join("\n")}
 </guest_context>
+${toolsNote}
 
-Return the briefing as a JSON object with a "guests" array containing structured information for each guest.`;
+For each guest listed above:
+1. Review their email and meeting history provided
+2. Use search tools to find their professional background
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/ai/meeting-briefs/generate-briefing.ts]: Already addressed - the prompt uses conditional language: 'If search tools are available, use them to research' (line 49) and 'SEARCH TIPS (if search tools are available)' (line 52).
